sql/core/src/main/scala/org/apache/spark/sql/catalyst/SQLBuilder.scala

```diff
@@ -38,14 +39,20 @@ import org.apache.spark.sql.types.{ByteType, DataType, IntegerType, NullType}
  * representations (e.g. logical plans that operate on local Scala collections), or are simply not
  * supported by this builder (yet).
  */
-class SQLBuilder(logicalPlan: LogicalPlan) extends Logging {
+class SQLBuilder(
```

Comment:
Let's mark this constructor as private and add a new public constructor without `nextSubqueryId`, `nextGenAttrId`, or `exprIdMap` since these arguments are only used within `SQLBuilder`.